backend/public-graph/graph/resolver.go

```diff
@@ -2680,23 +2681,30 @@ func (r *Resolver) ProcessPayload(ctx context.Context, sessionSecureID string, e
 
 			traceString := string(traceBytes)
 
+			serviceVersion := ""
+			if sessionObj.AppVersion != nil {
+				serviceVersion = *sessionObj.AppVersion
+			}
+
 			errorToInsert := &model.ErrorObject{
-				ProjectID:    projectID,
-				SessionID:    &sessionID,
-				Environment:  sessionObj.Environment,
-				Event:        v.Event,
-				Type:         v.Type,
-				URL:          v.URL,
-				Source:       v.Source,
-				LineNumber:   v.LineNumber,
-				ColumnNumber: v.ColumnNumber,
-				OS:           sessionObj.OSName,
-				Browser:      sessionObj.BrowserName,
-				StackTrace:   &traceString,
-				Timestamp:    v.Timestamp,
-				Payload:      v.Payload,
-				RequestID:    nil,
-				IsBeacon:     isBeacon,
+				ProjectID:      projectID,
+				SessionID:      &sessionID,
+				Environment:    sessionObj.Environment,
+				Event:          v.Event,
+				Type:           v.Type,
+				URL:            v.URL,
+				Source:         v.Source,
+				LineNumber:     v.LineNumber,
+				ColumnNumber:   v.ColumnNumber,
+				OS:             sessionObj.OSName,
+				Browser:        sessionObj.BrowserName,
+				StackTrace:     &traceString,
+				Timestamp:      v.Timestamp,
+				Payload:        v.Payload,
+				RequestID:      nil,
+				IsBeacon:       isBeacon,
+				ServiceVersion: serviceVersion,
+				ServiceName:    sessionObj.ServiceName,
```

Comment:
Does this differ from `v.ServiceName` like for backend errors?

Reply:
Good question! The payload is actually different for frontend errors, they are `ErrorObjectInput` and have no concept of a service 

![Screenshot 2023-08-30 at 11 12 26 AM](https://github.com/highlight/highlight/assets/58678/43a91784-3f7a-4685-85ec-b6134ea85f84)

At this point in the code path, we're processing a session payload -- the service name and version is not available as an input here because the client isn't sending it. 

The client passes service name and version as part of the initialize session endpoint:

https://github.com/highlight/highlight/blob/a3fcd94b95f6f9757871811a4c97f8d408ddaef5/backend/public-graph/graph/resolver.go#L1049-L1123

Hence, that's why it's pulled off the session object.